criu/cgroup.c

```diff
@@ -1242,7 +1237,9 @@ int prepare_task_cgroup(struct pstree_item *me)
 	 * in the cgset, and only unshare if that's true.
 	 */
 
-	return move_in_cgroup(se, !me->parent);
+	pr_info("avtni@ current_cgset: %d\n", current_cgset);
+
+	return move_in_cgroup(se, !parent_cg_set);
```

Comment:
It's hard to understand if the behavior here got changed or not. It would be clearer if you just passed another argument like `is_root_task`.